system/datatypes/mailinglist.php

```diff
@@ -768,8 +762,40 @@ private function sendMailing($mailingId) {
 		$this->xml->saveAndUnlock();
 	}
 
+	/**
+	 * @return bool
+	 */
 	private function hasSender() {
-		return '' != $this->getDoc()->getAttribute('sender-name') && '' != $this->getDoc()->getAttribute('sender-email');
+		$senderData = $this->getSenderData();
+
+		return '' != $senderData['name'] && '' != $senderData['email'];
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getSenderData() {
+		$senderData = [
+			'name' => $this->getDoc()->getAttribute('sender-name'),
+			'email' => $this->getDoc()->getAttribute('sender-email'),
+		];
+		if ('' == $senderData['name']) {
+			$senderData['name'] = hypha_getTitle();
+		}
+		if ('' == $senderData['email']) {
+			$senderData['email'] = hypha_getEmail();
+		}
+
+		return $senderData;
+	}
+
+	/**
+	 * @todo [LRM]: move so it can be used throughout Hypha
+	 *
+	 * @return string
+	 */
+	private function generateUniqueKey() {
+		return bin2hex(openssl_random_pseudo_bytes(8));
```

Comment:
This does not necessarily make the key unique, randomness just makes it really unlikely. A loop (like in `generateId()`) could be added here, but that's tricky to write generically (e.g. what's the scope of the uniqueness?). So I'm not sure if I like this change, it might be better to just keep an id for each address?